rpcs3/Emu/System.cpp

```diff
@@ -2941,15 +2941,21 @@ void Emulator::GracefulShutdown(bool allow_autoexit, bool async_op, bool savesta
 		return;
 	}
 
+	sys_log.notice("The game was requested to exit. Waiting for its reaction...");
+
 	auto perform_kill = [read_counter, allow_autoexit, this, info = GetEmulationIdentifier()]()
 	{
 		bool read_sysutil_signal = false;
 
-		u32 i = 100;
+		// If EXITGAME signal is not read, force kill after a couple of seconds.
```

Comment:
Hmm it's odd because games call cellSysutilCheckCallback every frame or so.
Maybe the read notification is broken.

Reply:
Alternatively we could add a sys_process_exit counter instead and remove the callback counter

Reply:
But maybe the game needs 2 seconds for other stuff. So it seems like this is the better approach ?

Reply:
The game needs more than a second to call sys_process_exit after the callback that triggers it.
And then it needs more than a second to stop.
That's why we need to increase the counter before we call the callback.
Otherwise our loop reaches its timeout of 2 seconds and kills the emu before the game terminated properly.

Reply:
The callback counter is there because when there is a hang, its likely that the callback would be missed entirely and we would waste time waiting for it.
We can, however, add a third timeout to sys_process_exit in case there is an issue in the terminaton process of the game.

Reply:
It's too late if we use the sys_process_exit as notifier.
The game will do time-consuming tasks before exiting.
We could of course use it to add even more time if that's necessary